optuna/study.py

```diff
@@ -707,17 +767,17 @@ def objective(trial):
         else:
             raise
 
+    if sampler is None and isinstance(direction_obj, Sequence):
```

Comment:
How about introducing a property held by each sampler that tells what dimensions they support? Since the dimensionality is specified at study construction, we could raise early errors here when sampler (and pruners) aren't compatible. It might also make sampler implementations easier since they've already communicated what dimensions they support and can assume that values or compatible with those.

Reply:
Looks good to me. I will follow your proposal.

Reply:
However, I think the dimension of the output, that is, `n_objectives`, may change depending on each trial, such as in conditional search space. What do you think?